modules/textures/src/lib/parsers/basis-module-loader.ts

```diff
@@ -43,3 +43,56 @@ function initializeBasisModule(BasisModule, wasmBinary) {
     });
   });
 }
+
+let loadBasisEncoderPromise;
+
+export async function loadBasisEncoderModule(options) {
+  const modules = options.modules || {};
+  if (modules.basisEncoder) {
+    return modules.basisEncoder;
+  }
+
+  loadBasisEncoderPromise = loadBasisEncoderPromise || loadBasisEncoder(options);
+  return await loadBasisEncoderPromise;
+}
+
+async function loadBasisEncoder(options) {
+  let BASIS_ENCODER = null;
+  let wasmBinary = null;
+
+  // TODO: load from `libs` path as soon as module is published on npm
+  [BASIS_ENCODER, wasmBinary] = await Promise.all([
+    await loadLibrary(
+      'https://raw.githubusercontent.com/BinomialLLC/basis_universal/master/webgl/encoder/build/basis_encoder.js',
+      null,
+      options
+    ),
+    await loadLibrary(
+      'https://raw.githubusercontent.com/BinomialLLC/basis_universal/master/webgl/encoder/build/basis_encoder.wasm',
+      null,
+      options
+    )
+  ]);
+
+  // Depends on how import happened...
+  // @ts-ignore TS2339: Property does not exist on type
+  BASIS_ENCODER = BASIS_ENCODER || globalThis.BASIS;
+  return await initializeBasisEncoderModule(BASIS_ENCODER, wasmBinary);
+}
+
+function initializeBasisEncoderModule(BasisEncoderModule, wasmBinary) {
```

Comment:
JSDoc on all functions. Mostly interested in a line that describes in word what the function does.

Reply:
Done